pkg/kubelet/kubelet_pods.go

```diff
@@ -1546,11 +1546,21 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 				continue
 			}
 		}
+
+		status := statuses[container.Name]
+		expectedHash := kubecontainer.HashContainer(&container)
+		containerStatus := podStatus.FindContainerStatusByName(container.Name)
+		if containerStatus != nil && containerStatus.Hash != expectedHash {
+			status.LastTerminationState = status.State
+			status.State = v1.ContainerState{Waiting: &v1.ContainerStateWaiting{Reason: "ContainerCreating"}}
```

Comment:
If I'm reading the [linked](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_pods.go#L1491) part of this function correctly, the default status for all containers is `ContainerCreating`. So I'm a little confused about why we need to be explicitly setting the `ContainerState` to the default value? Is it getting overwritten in another location?

Perhaps one helpful next step could be writing a test case that reproduces the error you're looking to fix?

Thanks :)

Reply:
At the first time, the container status is ContainerCreating. But if the image is changed, the container status will not change until the last container stoped. 
In my opinion, the container status should be containercreating between the time when image is changed and the time when last container stoped.

Reply:
Hmm, I apologize, I'm still not positive that I'm following.

Could you please write an unit test which reproduces the problem? I think that's our best path forward. Thanks!

Reply:
I do not know how to add a test for this feature. 
I will give a state transition description for this. It may help to understand why make this PR.